diff --git a/src/Controller/ForceLoginController.php b/src/Controller/ForceLoginController.php
index ee52e2a65..90dba6e2b 100644
--- a/src/Controller/ForceLoginController.php
+++ b/src/Controller/ForceLoginController.php
@@ -57,7 +57,9 @@ public function forceLogin() {
     $service_url_query_params = $this->requestStack->getCurrentRequest()->query->all();
     $cas_redirect_data = new CasRedirectData($service_url_query_params);
     $cas_redirect_data->setIsCacheable(TRUE);
-    return $this->casRedirector->buildRedirectResponse($cas_redirect_data, TRUE);
+    $response = $this->casRedirector->buildRedirectResponse($cas_redirect_data, TRUE);
+    $response->send();
+    return [];
   }
 
 }
