diff --git a/src/ContextServiceProvider.php b/src/ContextServiceProvider.php
deleted file mode 100644
index 96aeebf..0000000
--- a/src/ContextServiceProvider.php
+++ /dev/null
@@ -1,23 +0,0 @@
-<?php
-
-namespace Drupal\context;
-
-use Drupal\Core\DependencyInjection\ContainerBuilder;
-use Drupal\Core\DependencyInjection\ServiceProviderBase;
-
-/**
- * Alter the service container to use a custom class.
- */
-class ContextServiceProvider extends ServiceProviderBase {
-
-  /**
-   * {@inheritdoc}
-   */
-  public function alter(ContainerBuilder $container) {
-    // Override the menu active trail with a new class.
-    $definition = $container->getDefinition('menu.active_trail');
-    $definition->setClass('Drupal\context\ContextMenuActiveTrail');
-    $definition->addArgument($container->getDefinition('context.manager'));
-  }
-
-}
