diff --git a/field_group.module b/field_group.module
index e85c489..9e39102 100644
--- a/field_group.module
+++ b/field_group.module
@@ -174,7 +174,7 @@ function field_group_form_alter(array &$form, FormStateInterface $form_state) {
       );
 
       field_group_attach_groups($form, $context);
-      $form['#pre_render'][] = 'field_group_form_pre_render';
+      $form['#process'][] = 'field_group_form_pre_render';
     }
   }
 
diff --git a/src/Plugin/field_group/FieldGroupFormatter/DetailsSidebar.php b/src/Plugin/field_group/FieldGroupFormatter/DetailsSidebar.php
index e69de29..541b6af 100644
--- a/src/Plugin/field_group/FieldGroupFormatter/DetailsSidebar.php
+++ b/src/Plugin/field_group/FieldGroupFormatter/DetailsSidebar.php
@@ -0,0 +1,61 @@
+<?php
+
+/**
+ * /**
+ * @file
+ * Contains \Drupal\field_group\Plugin\field_group\FieldGroupFormatter\Details.
+ */
+
+namespace Drupal\field_group\Plugin\field_group\FieldGroupFormatter;
+
+/**
+ * Details Sidebar element.
+ *
+ * @FieldGroupFormatter(
+ *   id = "details_sidebar",
+ *   label = @Translation("Details Sidebar"),
+ *   description = @Translation("Add a details sidebar element"),
+ *   supported_contexts = {
+ *     "form",
+ *     "view"
+ *   }
+ * )
+ */
+class DetailsSidebar extends Details {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preRender(&$element, $rendering_object) {
+    parent::preRender($element, $rendering_object);
+
+    $element += [
+      '#group' => 'advanced',
+    ];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsForm() {
+    $form = parent::settingsForm();
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function settingsSummary() {
+    $summary = parent::settingsSummary();
+    return $summary;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function defaultContextSettings($context) {
+    $defaults = parent::defaultContextSettings($context);
+    return $defaults;
+  }
+
+}
