diff --git a/src/FlagCountManager.php b/src/FlagCountManager.php
index c2fa819..1f9ee2d 100644
--- a/src/FlagCountManager.php
+++ b/src/FlagCountManager.php
@@ -208,8 +208,9 @@ class FlagCountManager implements FlagCountManagerInterface, EventSubscriberInte
       ])
       ->expression('count', 'count + :inc', [':inc' => 1])
       ->execute();
-
-    $this->resetLoadedCounts($entity, $flag);
+    if($entity){
+      $this->resetLoadedCounts($entity, $flag);
+    }
   }
 
   /**
@@ -242,8 +243,9 @@ class FlagCountManager implements FlagCountManagerInterface, EventSubscriberInte
       else {
         $flaggings_count[$flag_id][$entity_id]++;
       }
-
-      $this->resetLoadedCounts($flagging->getFlaggable(), $flagging->getFlag());
+      if($entity = $flagging->getFlaggable()){
+        $this->resetLoadedCounts($entity, $flagging->getFlag());
+      }
     }
 
     // Build a query that fetches the count for all flag and entity ID
