diff --git a/js/password_encrypt.js b/js/password_encrypt.js
index eacce5607..128727c78 100644
--- a/js/password_encrypt.js
+++ b/js/password_encrypt.js
@@ -8,15 +8,15 @@
       var pass;
       var cpass;
 
-      $('#user-login, #user-login-form').submit(function (event) {
+      $('form.user-login, form.user-login-form', context).submit(function (event) {
         pass = $('#edit-pass').val();
         if (pass !== '') {
-          cipher = CryptoJS.AES.encrypt(pass, passkey);
+          cipher = CryptoJS.AES.encrypt(pass, passkey).toString();
           $('#edit-pass').val(cipher);
         }
       });
 
-      $('#user-register-form, #user-form').submit(function (event) {
+      $('form.user-register-form, form.user-form', context).submit(function (event) {
         pass = $('#edit-pass-pass1').val();
         cpass = $('#edit-pass-pass2').val();
 
@@ -27,7 +27,7 @@
         }
 
         if (pass !== '') {
-          cipher = CryptoJS.AES.encrypt(pass, passkey);
+          cipher = CryptoJS.AES.encrypt(pass, passkey).toString();
           $('#edit-pass-pass1').val(cipher);
           $('#edit-pass-pass2').val(cipher);
         }
