diff --git a/modules/rdf/rdf.module b/modules/rdf/rdf.module
index 8d4329884d..933d6ff773 100644
--- a/modules/rdf/rdf.module
+++ b/modules/rdf/rdf.module
@@ -236,11 +236,13 @@ function rdf_comment_storage_load($comments) {
     /** @var \Drupal\comment\CommentInterface $comment*/
     $comment->rdf_data['date'] = rdf_rdfa_attributes($created_mapping, $comment->get('created')->first()->toArray());
     $entity = $comment->getCommentedEntity();
-    // The current function is a storage level hook, so avoid to bubble
-    // bubbleable metadata, because it can be outside of a render context.
-    $comment->rdf_data['entity_uri'] = $entity->toUrl()->toString(TRUE)->getGeneratedUrl();
-    if ($comment->hasParentComment()) {
-      $comment->rdf_data['pid_uri'] = $comment->getParentComment()->url();
+    if($entity){
+      // The current function is a storage level hook, so avoid to bubble
+      // bubbleable metadata, because it can be outside of a render context.
+      $comment->rdf_data['entity_uri'] = $entity->toUrl()->toString(TRUE)->getGeneratedUrl();
+      if ($comment->hasParentComment() && $comment->getParentComment()) {
+        $comment->rdf_data['pid_uri'] = $comment->getParentComment()->url();
+      }
     }
   }
 }
