diff --git a/src/RouteProvider.php b/src/RouteProvider.php
index aedd588..b5fb5f9 100644
--- a/src/RouteProvider.php
+++ b/src/RouteProvider.php
@@ -35,33 +35,13 @@ class RouteProvider extends CoreRouteProvider {
   /**
    * {@inheritdoc}
    */
-  public function getRouteCollectionForRequest(Request $request) {
-    // Cache both the system path as well as route parameters and matching
-    // routes.
+  protected function getRouteCollectionCacheId(Request $request) {
+    // Include the current language code in the cache identifier as
+    // the language information can be elsewhere than in the path, for example
+    // based on the domain.
+    $language_part = $this->getCurrentLanguageCacheIdPart();
     $workspace_id = $this->workspaceManager->getActiveWorkspace()->id();
-    $cid = 'route:' . "workspace$workspace_id:" . $request->getPathInfo() . ':' . $request->getQueryString();
-    if ($cached = $this->cache->get($cid)) {
-      $this->currentPath->setPath($cached->data['path'], $request);
-      $request->query->replace($cached->data['query']);
-      return $cached->data['routes'];
-    }
-    else {
-      // Just trim on the right side.
-      $path = $request->getPathInfo();
-      $path = $path === '/' ? $path : rtrim($request->getPathInfo(), '/');
-      $path = $this->pathProcessor->processInbound($path, $request);
-      $this->currentPath->setPath($path, $request);
-      // Incoming path processors may also set query parameters.
-      $query_parameters = $request->query->all();
-      $routes = $this->getRoutesByPath(rtrim($path, '/'));
-      $cache_value = [
-        'path' => $path,
-        'query' => $query_parameters,
-        'routes' => $routes,
-      ];
-      $this->cache->set($cid, $cache_value, CacheBackendInterface::CACHE_PERMANENT, ['route_match']);
-      return $routes;
-    }
+    return 'route:' . "workspace{$workspace_id}:" . $language_part . ':' . $request->getPathInfo() . ':' . $request->getQueryString();
   }
 
 }
